pkg/auth/permissions/resource.go

```diff
@@ -71,3 +78,19 @@ func WithLegacyAuthForSAC(md ResourceMetadata, use bool) ResourceMetadata {
 	md.legacyAuthForSAC = &use
 	return md
 }
+
+// CheckResourceForAccess will verify whether the given ResourceMetadata is contained within the map
+// with at least the specified storage.Access.
+// Note: This will take replacing resources into account.
+func CheckResourceForAccess(resource ResourceMetadata, resourceAccessMap map[string]storage.Access,
```

Comment:
Maybe take `ResourceMetadata` as a receiver and rename to `IsPermittedBy()`?

Reply:
I've made the changes within #1834 , let's move the discussion there, if there are any additional comments.